client/src/constants/forms.js

```diff
@@ -79,819 +58,122 @@ debug.enabled = false;
  * }}
  */
 
-/**
- * Gets the relevant fields of a given HMModelFlat object. For example,
- * if a form only uses the `params` field of the HMModelFlat, and 2 other
- * root-level fields, it will only return the model name field, the params
- * field, and those 2 other fields.
- *
- * @typedef GetRelevantHMModelFlatFunction
- * @type {(hmModelFlat: HMModelFlat) => object}
- */
-
-/**
- * Builds the `modelChoicesData` based on the given `hmModelFlat`.
- *
- * Typically this function will take what exists in `hmModelFlat` and transpose
- * the relevant fields onto the object which corresponds to the currently
- * chosen model.
- *
- * @typedef GetModelChoicesDataFromFlatFunction
- * @type {(hmModelFlat: HMModelFlat) => ModelChoicesData}
- */
+ /**
+  * Defines the parameters associated with the subform common to all model
+  * selections associated with that subform.
+  * @typedef CoreParams
+  * @type {Array.String}
+  */
 
-/**
- * Holds the different data + paremeters associated with the model choices.
- * The use of this object and it's objects change based on the form and on
- * the `getModelChoicesDataFromFlat` function, but the structure below is
- * consistent among all forms.
- *
- * @typedef ModelChoicesData
- * @type {{
- *  [modelName: string]: object
- * }}
- */
-
-/**
- * Updates the given `hmModelFlat` and `modelChoicesData` based on the provided
- * `oldModelName` and `newModelName`. This funciton allows a form to customize
- * how the HMModelFlat is modified when the selected model is changed, as well
- * as the current ModelChoicesData.
- *
- * Most of the time, the params field for the form is modified when this
- * happens, but custom behavior is needed for several forms which make use of
- * root-level fields.
- *
- * @typedef UpdateModelChoiceFunction
- * @type {(
- *  oldModelName: string,
- *  newModelName: string,
- *  hmModelFlat: HMModelFlat,
- *  modelChoicesData: ModelChoicesData
- * ) => [HMModelFlat, ModelChoicesData]}
- */
-
-/**
- * Flattens the given HMModel's parameters which are relevant to the form into
- * a format usable by the server.
- *
- * For example, this is used to convert an `HMModel` from its
- * expanded form, into an `HMModelFlat` when done over all forms.
- *
- * @typedef FlattenHMModelFunction
- * @type {(hmModel: HMModel) => object}
- */
-
-/**
- * The primary source of information on the client side which will generate
- * the forms. This contains everything that is needed to configure the
- * front-end forms.
- *
- * @type {Forms}
- */
 const FORMS = {
```

Comment:
The reason for specifying the type is so that autocomplete works in other parts of the application that use this object.

Reply:
Added it. Was an accidental deletion.